### PR TITLE
docs: move c8y token docs and expand on details and alternatives

### PR DESCRIPTION
**Pull request by reubenmiller**
## Proposed changes

Updates to the Cumulocity IoT Token (JWT) documentation:

* Move page under the security section
* Add note about how the proxy service should be preferred over manual api calls
* Add more examples on how to retrieve the token and how to use it.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [x] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue

https://github.com/thin-edge/thin-edge.io/issues/2551

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [ ] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by gligorisaev — APPROVED** [review]

**Comment by didier-wenzek** [comment]
> :warning: This PR is on hold as there is an thin-edge.io example which still requires to fetch a jwt as it establishing a websocket with c8y directly (see [example](https://github.com/thin-edge/thin-edge.io_examples/tree/main/webrtc-webcam)). We need to check if the example can utilize the c8y auth proxy or not.

We can simply move this content under [Security and Access Control](https://thin-edge.github.io/thin-edge.io/operate/security/) with a note referencing the auth proxy and telling that in most cases this is not required.

**Comment by reubenmiller** [comment]
> > ⚠️ This PR is on hold as there is an thin-edge.io example which still requires to fetch a jwt as it establishing a websocket with c8y directly (see [example](https://github.com/thin-edge/thin-edge.io_examples/tree/main/webrtc-webcam)). We need to check if the example can utilize the c8y auth proxy or not.
> 
> We can simply move this content under [Security and Access Control](https://thin-edge.github.io/thin-edge.io/operate/security/) with a note referencing the auth proxy and telling that in most cases this is not required.

Yes that sounds like a better way to go...I'll update the PR

**Review by gligorisaev — APPROVED** [review]

File: docs/src/operate/security/cumulocity_token.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+# Get the Cumulocity IoT URL
+export C8Y_URL="https://$(tedge config get c8y.url)"
+
+# Get the token (using mosquitto_rr cli command)
+export C8Y_TOKEN=$(mosquitto_rr -t c8y/s/uat -e c8y/s/dat -m '' | cut -d, -f2-)
+
+# Send a REST API call to Cumulocity IoT
+curl -s -H "Authorization: Bearer $C8Y_TOKEN" \
+    -H "Accept: application/json" \
+    "$C8Y_URL/user/currentUser"
```
Comment (didier-wenzek) [inline docs/src/operate/security/cumulocity_token.md:102]: As a conclusion of this example, I would add the equivalent call using the Cumulocity Proxy. just to make the reader want.

```
curl http://localhost:8001/c8y/user/currentUser 
```

**Review by didier-wenzek — APPROVED** [review]
Approved with one suggestion.

File: docs/src/operate/security/cumulocity_token.md (hunk 1 above)
Comment (reubenmiller) [reply docs/src/operate/security/cumulocity_token.md:102]: Resolved by 93b12f47b181b42cd9709e8639f349722ec8d152

Comment (reubenmiller) [reply docs/src/operate/security/cumulocity_token.md:102]: Resolved by 93b12f47b181b42cd9709e8639f349722ec8d152

**Review by didier-wenzek — APPROVED** [review]
Approved

**Pull request merged**